DirectProgramming/C++SYCL_FPGA/include/annotated_class_util.hpp

```diff
@@ -0,0 +1,153 @@
+#ifndef __ANNOTATED_CLASS_UTIL_HPP__
+#define __ANNOTATED_CLASS_UTIL_HPP__
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl::ext::oneapi::experimental;
+
+namespace fpga_tools {
+///////////////////////////////////////////////////////////////////////////////
+// This header provides a utility function: "alloc_annotated", which improves
+// malloc_shared" in allocating host/share memory in the following aspects:
+//   (1) better code brevity
+//   (2) support compile-time type check
+//
+// "annotated_alloc" function takes an annotated_arg type as the only template
+// parameter, and returns an instance of that annotated_arg type. This provides
+// compile-time guarantee that the properties of the allocated memory (e.g.
+// buffer location, alignment) match with the annotations on the kernel arguments
+//
+// To use "alloc_annotated",
+//
+// 1. include header "annotated_class_util.hpp"
+// 2. create a type alias for the "annotated_arg" type, e.g.
+//
+//    using karg_t = sycl::ext::oneapi::experimental::annotated_arg<
```

Comment:
```suggestion
//    using annotated_arg_t = sycl::ext::oneapi::experimental::annotated_arg<
```

I don't like recommending `karg_t` because the `k` prefix is usually used to indicate a constant, and it is kind of confusing to read in code.

Reply:
Updated